core/src/main/java/com/google/zxing/pdf417/encoder/PDF417HighLevelEncoderTestAdapter.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 ZXing authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.zxing.pdf417.encoder;
+
+import com.google.zxing.WriterException;
+
+import java.nio.charset.Charset;
+
+/**
+ * Test adapter for PDF417HighLevelEncoder to be called solely from unit tests.
+ */
+
+public final class PDF417HighLevelEncoderTestAdapter {
+
+  private PDF417HighLevelEncoderTestAdapter() {
+  }
+
+  public static String encodeHighLevel(String msg, Compaction compaction, Charset encoding) throws WriterException {
```

Comment:
For now can this just be a utility method in your new test class?

Reply:
I am not sure that I understand the question. What are you referring to with "new test class"? 
The problem is that the unit test com.google.zxing.pdf417.decoder.PDF417DecoderTestCase cannot access the method com.google.zxing.pdf417.encoder.PDF417HighLevelEncoder.encodeHighLevel() because neither the class nor the method are public. I don't think that we should change that since the method is internal and not part of the public API. I therefore introduced the class com.google.zxing.pdf417.encoder.PDF417HighLevelEncoderTestAdapter that is a public proxy to that method. It contains a clear comment that hopefully keeps people from calling this method from their code. I could perhaps also add some code in the function that checks the call stack at runtime and refuses to to operate if it is called from anywhere else than from the unit test. The issue with that may be that it slows down the unit test since the called thousands of times from the unit test. I don't find this solution ideal but I don't have a better idea either. I am open to any other idea.

Reply:
This new class you added right here.
Maybe I'm not reading this right but all classes in question are in the same package, so package-private access should be OK to call.

Reply:
The test itself is in the package com.google.zxing.pdf417.decoder and it needs to call a method from com.google.zxing.pdf417.encoder.

Reply:
Exactly, that's a good summary of the problem.